processes-endpoint.md

```diff
@@ -140,32 +140,144 @@ In a future version, a solution supporting a live-feed can also be included.
+          "href": "/api/core/bitstreams/e2343390-0b1b-4397-be3d-66fed243a436/format"
+        },
+        "self": {
+          "href": "/api/core/bitstreams/e2343390-0b1b-4397-be3d-66fed243a436"
+        }
+      },
+      "_embedded": {
+        "format": {
+          "id": 1,
+          "shortDescription": "CSV",
+          "description": "Comma-separated values",
+          "mimetype": "text/csv",
+          "supportLevel": 0,
+          "internal": false,
+          "extensions": null,
+          "type": "bitstreamformat",
+          "_links" : {
+            "self" : {
+              "href": "/api/core/bitstreamformats/27"
+            }
+          }
+        }
+      }
     },
     "mapfile" : {
-      "href" : "/api/system/processes/4/files/mapfile"
+      "id": "e2343390-0b1b-4397-be3d-66fed243a436",
+      "uuid": "e2343390-0b1b-4397-be3d-66fed243a436",
+      "name": "map.txt",
+      "handle": null,
+      "metadata": {
+        "dc.title": [
+          {
+            "value": "map.txt",
+            "language": null,
+            "authority": null,
+            "confidence": -1,
+            "place": 0
+          }
+        ],
+        "process.type": [
+          {
+            "value": "mappingFile",
+            "language": null,
+            "authority": null,
+            "confidence": -1,
+            "place": 0
+          }
+        ]
+      },
+      "bundleName": null,
+      "sizeBytes": 479381,
+      "checkSum": {
+        "checkSumAlgorithm": "MD5",
+        "value": "64cdb4b4dd01b4a4cbb41e6d879fe996"
+      },
+      "sequenceId": null,
+      "type": "bitstream",
+      "_links": {
+        "content": {
+          "href": "/api/core/bitstreams/e2343390-0b1b-4397-be3d-66fed243a436/content"
+        },
+        "format": {
+          "href": "/api/core/bitstreams/e2343390-0b1b-4397-be3d-66fed243a436/format"
+        },
+        "self": {
+          "href": "/api/core/bitstreams/e2343390-0b1b-4397-be3d-66fed243a436"
+        }
     },
-    "zipfile" : {
-      "href" : "/api/system/processes/4/files/zipfile"
+      "_embedded": {
+        "format": {
+          "id": 1,
+          "shortDescription": "Text",
+          "description": "Plain Text",
+          "mimetype": "text/plain",
+          "supportLevel": 0,
+          "internal": false,
+          "extensions": null,
+          "type": "bitstreamformat",
+          "_links": {
+            "self": {
+              "href": "/api/core/bitstreamformats/6"
+            }
+          }
+        }
+      }
     }
   }
 }
 ```
 
-## Execution File Output
-**GET /api/system/processes/<:process-id>/files/<:file-name>**
-
-This endpoint will let an administrator download an output file created by a process. If the file is found, it will be presented as a download. This endpoint will support "Range" HTTP headers so that downloads can be paused and resumed.
+## Execution File Output List (type filter)
+**GET /api/system/processes/<:process-id>/files/<:type>**
```

Comment:
I recommend to avoid so deeply nested endpoint, this is a search method over the bitstream endpoint by Process and Type

Reply:
The files are a sub-resource of the process, so there's nothing too deeply nested. It is very similar to other endpoints where 2 IDs are used.
A search of bitstreams would need the types to be known by Angular (requiring an endpoint such as https://github.com/DSpace/Rest7Contract/pull/99/files#r435198271). This would add quite some overhead just to use a search for exposing a sub-resource while the sub-resource syntax is appropriate here